libgalois/src/PerThreadStorage.cpp

```diff
@@ -38,15 +38,21 @@ galois::substrate::PerBackend& galois::substrate::getPPSBackend() {
 }
 
 #define MORE_MEM_HACK
+#ifdef GALOIS_LOW_MEM_FOOTPRINT
+// used for CI testing and other situations where galois memory
+// footprint needs to be low
+const size_t allocSize = 1 * (2 << 20); // 2MB
```

Comment:
There should be a nicer way to configure this rather than requiring a complete rebuild of the library.

I don't know the history of `MORE_MEM_HACK` but maybe a simple solution would be to just return to that code, which just allocates 4MB at a time, and file a ticket to verify later that there aren't any important performance regressions.

Reply:
That is also fine with me. Not sure if anyone knows where that MORE_MEM_HACK came from (I don't); some context would be appreciated from anyone who does know.

Reply:
Looking at the code, I would guess that it was added because certain applications exhausted the buffer used to store per-thread data.

Correcting myself above, it looks like this is only a correctness issue (not a performance one) and libgalois will die if allocSize is too small, so it should be pretty straightforward to confirm if any apps actually require the larger allocSize and then determine if it is a true capacity problem or if we have too much fragmentation or internal leakage.

Reply:
Interesting; so it actually never allocates more than that.

PerThread vectors and stuff of the sort might depend on the graph being used (e.g., amount you push to a vector depends on your input), so it may not be completely straightforward, but more static uses of PPS like accumulators and such can be easily tested with CI.

Reply:
I don't think there is an allocator that draws from the PerThread/PerSocket allocation pool so any PerThread<vector> like things would either use malloc or runtime/Mem.h (if they happened to use `gstl::Vector`. The memory for vector metadata (size, pointer to array) would be drawn from the PerThread allocation pool but that should be pretty small relative to the dynamic array itself.

Reply:
I see.

I can remove both mem hacks and use the huge page per thread; I'll try that in the next commit that will include Roshan's hanging fix.